frontend/dockerfile/dockerignore/dockerignore_test.go

```diff
@@ -1,61 +1,54 @@
 package dockerignore
 
 import (
-	"os"
-	"path/filepath"
+	"strings"
 	"testing"
 )
 
 func TestReadAll(t *testing.T) {
-	di, err := ReadAll(nil)
+	actual, err := ReadAll(nil)
 	if err != nil {
-		t.Fatalf("Expected not to have error, got %v", err)
+		t.Errorf("Expected no error, got %v", err)
 	}
-
-	if diLen := len(di); diLen != 0 {
-		t.Fatalf("Expected to have zero dockerignore entry, got %d", diLen)
+	if entries := len(actual); entries != 0 {
+		t.Fatalf("Expected to have zero entries, got %d", entries)
 	}
 
-	diName := filepath.Join(t.TempDir(), ".dockerignore")
-	content := "test1\n/test2\n/a/file/here\n\nlastfile\n# this is a comment\n! /inverted/abs/path\n!\n! \n"
-	err = os.WriteFile(diName, []byte(content), 0600)
-	if err != nil {
-		t.Fatal(err)
-	}
+	const content = `test1
+/test2
+/a/file/here
 
-	diFd, err := os.Open(diName)
-	if err != nil {
-		t.Fatal(err)
+lastfile
+# this is a comment
+! /inverted/abs/path
+!
+! `
+
+	expected := []string{
+		"test1",
+		"test2",       // according to https://docs.docker.com/engine/reference/builder/#dockerignore-file, /foo/bar should be treated as foo/bar
+		"a/file/here", // according to https://docs.docker.com/engine/reference/builder/#dockerignore-file, /foo/bar should be treated as foo/bar
+		"lastfile",
+		"!inverted/abs/path",
+		"!",
+		"!",
 	}
-	defer diFd.Close()
 
-	di, err = ReadAll(diFd)
+	actual, err = ReadAll(strings.NewReader(content))
 	if err != nil {
-		t.Fatal(err)
+		t.Error(err)
 	}
 
-	if len(di) != 7 {
```

Comment:
I think this whole block can just be simplified to be `require.Equal(t, expected, di)`?

Similarly above, we should just `require.NoError` - there shouldn't be any reason we can't use testify in this package?

Reply:
I looked at https://github.com/moby/patternmatcher/blob/main/go.mod and it currently didn't have any dependencies, so tried to avoid introducing one (if not _really_ needed).

Reply:
Yeah, honestly it wasn't a "great" test to start with, so I also didn't want to spend too much time on it (for now at least).